File: 1.fizzbuzz/js/src/fizzbuzz.js
exports.getResult = function(num) {
    var pattern = {
        3: 'fizz',
        5: 'buzz',
    }

    return Object.keys(pattern).reduce(function(p, k) {
        return p + (num % k === 0 ? pattern[k] : '')
    }, '') || num
}

File: 1.fizzbuzz/scala/src/main/scala/FizzBuzz.scala

object FizzBuzz {
  def getResult(num: Int) = {
    (num % 3, num % 5) match {
      case (0, 0) => "fizzbuzz"
      case (0, _) => "fizz"
      case (_, 0) => "buzz"
      case _ => num
    }
  }
}
